homeassistant/components/ambient_station/binary_sensor.py

```diff
@@ -1,34 +1,208 @@
 """Support for Ambient Weather Station binary sensors."""
 from __future__ import annotations
 
+from dataclasses import dataclass
+
 from homeassistant.components.binary_sensor import (
-    DOMAIN as BINARY_SENSOR,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_CONNECTIVITY,
     BinarySensorEntity,
+    BinarySensorEntityDescription,
 )
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import ATTR_NAME
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from . import (
-    SENSOR_TYPES,
-    TYPE_BATT1,
-    TYPE_BATT2,
-    TYPE_BATT3,
-    TYPE_BATT4,
-    TYPE_BATT5,
-    TYPE_BATT6,
-    TYPE_BATT7,
-    TYPE_BATT8,
-    TYPE_BATT9,
-    TYPE_BATT10,
-    TYPE_BATT_CO2,
-    TYPE_BATTOUT,
-    TYPE_PM25_BATT,
-    TYPE_PM25IN_BATT,
-    AmbientWeatherEntity,
+from . import AmbientWeatherEntity
+from .const import ATTR_LAST_DATA, DATA_CLIENT, DOMAIN
+
+TYPE_BATT1 = "batt1"
+TYPE_BATT10 = "batt10"
+TYPE_BATT2 = "batt2"
+TYPE_BATT3 = "batt3"
+TYPE_BATT4 = "batt4"
+TYPE_BATT5 = "batt5"
+TYPE_BATT6 = "batt6"
+TYPE_BATT7 = "batt7"
+TYPE_BATT8 = "batt8"
+TYPE_BATT9 = "batt9"
+TYPE_BATT_CO2 = "batt_co2"
+TYPE_BATTOUT = "battout"
+TYPE_PM25_BATT = "batt_25"
+TYPE_PM25IN_BATT = "batt_25in"
+TYPE_RELAY1 = "relay1"
+TYPE_RELAY10 = "relay10"
+TYPE_RELAY2 = "relay2"
+TYPE_RELAY3 = "relay3"
+TYPE_RELAY4 = "relay4"
+TYPE_RELAY5 = "relay5"
+TYPE_RELAY6 = "relay6"
+TYPE_RELAY7 = "relay7"
+TYPE_RELAY8 = "relay8"
+TYPE_RELAY9 = "relay9"
+
+
+@dataclass
+class AmbientBinarySensorDescriptionMixin:
+    """Define an entity description mixin for binary sensors."""
+
+    on_state: int
```

Comment:
If you add a default to `on_state`, you could add it directly to `AmbientBinarySensorDescription`.
There is something to having it as required field as well. Up to you how you like it.

Reply:
Got it. I think I'll stick with having to explicitly define it.